src/platform/silabs/wifi/WifiInterface.h

```diff
@@ -148,11 +140,43 @@ typedef enum
 } sl_wfx_interface_t;
 #endif
 
+// TODO: Figure out if we need this structure. We have different strcutures for the same use
+struct WifiCredentials
+{
+    std::array<uint8_t, WFX_MAX_SSID_LENGTH> ssid       = { 0 };
```

Comment:
It looks like the only functionality of std::array used in here is fill(0),  which could also be performed by a memset.
The use of `std` usually leads to a code size increase, could we have a comparison on the impact? 

I am all for the usage of the std library but as we are constantly fighting for memory I am wondering if chasing all the std to convert to native C functions will not end up being an optimization we will do in the future to save on code size.

Reply:
reverted back to C-like arrays